### PR TITLE
Admin UI: Fixes for "Client details page" that were not included in the previous branch due to premature merge

### PR DESCRIPTION
**Pull request by NikolaVetnic**
# Readiness checklist

-   [X] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [X] I ensured that the PR title is good enough for the changelog.
-   [X] I labeled the PR.

# Description

I addressed the issues regarding the Backbone part of the task left over accidentally merging the #776 PR.

### CODE REVIEW

**Review by jkoenig134 — DISMISSED** [review]
Giving my approval for die aui part.

File: Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs [hunk 1]
```diff
@@ -6,6 +6,5 @@ public class ClientInfo
     public required string DisplayName { get; set; }
     public required string DefaultTier { get; set; }
     public required DateTime CreatedAt { get; set; }
-    public required int NumberOfIdentities { get; set; }
```
Comment (tnotheis) [inline Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs:9]: I don't get it. Why did you remove `NumberOfIdentities` now?

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs [hunk 2]
```diff
@@ -33,6 +38,16 @@ public async Task<IEnumerable<OAuthClient>> FindAll(CancellationToken cancellati
         return oAuthClients;
     }
 
+    public async Task<Dictionary<OAuthClient, int>> CountIdentities(List<OAuthClient> clients, CancellationToken cancellationToken, bool track = false)
```
[the remark this answers is not in the document]
Comment (NikolaVetnic) [reply Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs:41]: Daniel Almeida informed me about this request in chat. He explicitly said "...talked to Timo..." and "...it receives a list of clients and retrieves the number of identities for each...". That is why I did it like so.

I agree the list of IDs is enough.

File: Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs [hunk 3]
```diff
@@ -33,6 +38,16 @@ public async Task<IEnumerable<OAuthClient>> FindAll(CancellationToken cancellati
...
+    {
+        var result = new Dictionary<OAuthClient, int>();
+
+        foreach (var client in clients)
+            result[client] = await _readonlyIdentities.CountAsync(i => i.ClientId == client.ClientId, cancellationToken);
```
[the remark this answers is not in the document]
Comment (NikolaVetnic) [reply Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs:46]: Changed.

File: Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs (hunk 1 above)
Comment (NikolaVetnic) [reply Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs:9]: By mistake. I did not notice this while manually testing since it only appears when SDK is used, and I didn't check responses in the tests.

Comment (tnotheis) [reply Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs:9]: That's why I keep saying "review your own changes before you commit them" (this is just a reminder - of course you miss something from time to time)

File: Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs [hunk 4]
```diff
@@ -69,6 +69,7 @@ public async Task ThenTheResponseContainsAClient()
         _response!.Result!.ClientId.Should().Be(_clientId);
         _response!.Result!.DefaultTier.Should().Be(_tierId);
         _response!.Result!.MaxIdentities.Should().Be(_maxIdentities);
+        _response!.Result!.NumberOfIdentities.Should().BeLessOrEqualTo(_maxIdentities);
```
Comment (tnotheis) [inline Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs:72]: This is repetition of business logic. Also, it's logic that isn't even related to this test.

I will remove it.

**Review by tnotheis — APPROVED** [review]

File: Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs (hunk 4 above)
Comment (NikolaVetnic) [reply Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs]: ...that is why I added it here, so that it makes life easier for anyone in the future who may be in the dilemma whether it is necessary or not.

File: Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs (hunk 1 above)
Comment (NikolaVetnic) [reply Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs:9]: I did review it. I got the response as expected when I manually tested with a client (Insomnia). But the problem was that the `ClientInfo` is only used by the SDK which is (at the moment) only used by the integration tests. That is why it slipped by, it wasn't checked by anything if it were there, and so... [continued](https://github.com/nmshd/backbone/pull/797#discussion_r1715220005)

Comment (tnotheis) [reply Sdks/AdminApi.Sdk/src/Endpoints/Clients/Types/ClientInfo.cs:9]: > I got the response as expected when I manually tested with a client (Insomnia)

I'm not talking about manual testing, I'm talking about looking at the code changes before you `git stage` them. Such a review usually uncovers a problem like accidentally deleting something.

File: Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs (hunk 4 above)
Comment (tnotheis) [reply Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs]: I see. 

I would expect our schema validator to uncover such problems (=> additional properties that don't exist on the type). Since that seems to not be the case, we should rather fix it there.

Alternatively, as a short-term solution, we could have checked with `.Should().NotBeNull()`. This lacks any business logic and still makes sure that what you describe cannot happen.
But let's not do that now, as this would trigger another pipeline run. And I finally want to merge this. :)
I will add the improvement of the schema validator to our backlog.

**Review by jkoenig134 — APPROVED** [review]
